src/util/adb.js

```diff
@@ -210,10 +210,45 @@ export default class ADBUtils {
     await this.runShellCommand(deviceId, ['mkdir', '-p', artifactsDir]);
 
     this.artifactsDirMap.set(deviceId, artifactsDir);
-
     return artifactsDir;
   }
 
+  async checkOrCleanArtifacts(
+    deviceId: string, clean: boolean): Promise<boolean> {
+    const {adbClient} = this;
+
+    var directories = false;
+
+    await wrapADBCall(async () => {
+      log.debug('Finding Older Artifacts');
+
+      const dirs = await adbClient.readdir(deviceId, '/sdcard');
+      for (var dir = dirs.length - 1; dir >= 0; --dir ) {
```

Comment:
I'm not really sure that we care about the order of the directories that much, if we really do we could still sort them before iterating over them.

I would just iterate over the list we got in the order we got it, with the "lighter" js syntax possible:
```
for (let file of files) {

}
```